src/saveload/afterload.cpp

```diff
@@ -1008,7 +1008,7 @@ bool AfterLoadGame()
 					if ((GB(t.m5(), 4, 2) == ROAD_TILE_CROSSING ? (Owner)t.m3() : GetTileOwner(t)) == OWNER_TOWN) {
 						SetTownIndex(t, CalcClosestTownFromTile(t)->index);
 					} else {
-						SetTownIndex(t, 0);
```

Comment:
it's not immediately clear to me why TOWN_BEGIN is used without iterating to TOWN_END - it almost looks like "TOWN_INVALID" but that's distinct, so apparently not? Seems like it needs some documenting? there's no doxygen on the definition

Reply:
So you're saying 'TOWN_BEGIN` is badly named, right? I named it `_BEGIN` so it's somewhat consistent with containers in C++ (`.begin()`).

What would be a better name for the town with ID 0?

Note that "first" is also not quite right, because there might not be a town with ID 0 (removal of that town, or in scenario editor without towns at all). 
There is `.front()` on containers, but that's UB when there's nothing in the container.

Solving that potential issue with ID 0 is not something I'm going to be doing. Alternatively I'll change  this to `static_cast<TownID>(0)` and keep the magic number in there.